src/Bicep.Core/Semantics/SymbolExtensions.cs

```diff
@@ -7,24 +7,10 @@ namespace Bicep.Core.Semantics
 {
     public static class SymbolExtensions
     {
-        public static SyntaxBase? SafeGetBodyPropertyValue(this ResourceSymbol resourceSymbol, string propertyName)
-        {
-            if (resourceSymbol.DeclaringResource.Body is not ObjectSyntax body)
-            {
-                return null;
-            }
+        public static SyntaxBase? SafeGetBodyPropertyValue(this ResourceSymbol resourceSymbol, string propertyName) => 
```

Comment:
>SafeGetBodyPropertyValue [](start = 34, length = 24)

Our usage pattern doesn't seem to care whether a resource/module is conditional or not, so I kept the behavior.